pkg/apis/config/sigstore_keys.go

```diff
@@ -0,0 +1,184 @@
+)
+
+const (
+	// SigstoreKeysConfigName is the name of ConfigMap created by the
+	// reconciler and consumed by the admission webhook for determining
+	// which Keys/Certificates are trusted for things like Fulcio/Rekor, etc.
+	SigstoreKeysConfigName = "config-sigstore-keys"
+)
+
+// Note that these are 1:1 mapped to public API SigstoreKeys. Reasoning
+// being that we may choose to serialize these differently, or use the protos
+// that are defined upstream, so want to keep the public/private distinction, so
+// that we can change things independend of breaking the API. Time will tell
+// if this is the right call, but we can always reunify them later if we so
+// want.
+
+// TransparencyLogInstance describes the immutable parameters from a
+// transparency log.
+// See https://www.rfc-editor.org/rfc/rfc9162.html#name-log-parameters
+// for more details.
+// The incluced parameters are the minimal set required to identify a log,
+// and verify an inclusion promise.
+type TransparencyLogInstance struct {
+	BaseURL       apis.URL `json:"baseURL"`
+	HashAlgorithm string   `json:"hashAlgorithm"`
+	// PEM encoded public key
+	PublicKey []byte `json:"publicKey"`
+	LogID     string `json:"logID"`
+}
+
+type DistinguishedName struct {
+	Organization string `json:"organization"`
+	CommonName   string `json:"commonName"`
+}
+
+type CertificateAuthority struct {
+	// The root certificate MUST be self-signed, and so the subject and
+	// issuer are the same.
+	Subject DistinguishedName `json:"subject"`
+	// The URI at which the CA can be accessed.
+	URI apis.URL `json:"uri"`
+	// The certificate chain for this CA.
+	// CertChain is in PEM format.
+	CertChain []byte `json:"certChain"`
+
+	// TODO(vaikas): How to best represent this
+	// The time the *entire* chain was valid. This is at max the
+	// longest interval when *all* certificates in the chain where valid,
+	// but it MAY be shorter.
+	//       dev.sigstore.common.v1.TimeRange valid_for = 4;
+}
+
+// SigstoreKeys contains all the necessary Keys and Certificates for validating
+// against a specific instance of Sigstore.
+// TODO(vaikas): See about replacing these with the protos here once they land
+// and see how easy it is to replace with protos instead of our custom defs
+// above.
+// https://github.com/sigstore/protobuf-specs/pull/5
+// And in particular: https://github.com/sigstore/protobuf-specs/pull/5/files#diff-b1f89b7fd3eb27b519380b092a2416f893a96fbba3f8c90cfa767e7687383ad4R70
+// Well, not the multi-root, but one instance of that is exactly the
+// SigstoreKeys.
+type SigstoreKeys struct {
+	// Trusted certificate authorities (e.g Fulcio).
+	CertificateAuthority []CertificateAuthority `json:"certificateAuthority"`
+	// Rekor log specifications
+	TLog []TransparencyLogInstance `json:"tLog"`
+	// Certificate Transparency Log
+	CTLog []TransparencyLogInstance `json:"ctLog"`
+	// Trusted timestamping authorities
+	TimeStampAuthorities []CertificateAuthority `json:"timestampAuthorities"`
```

Comment:
In the protos definition, this appears as `timestamp_authorities`

Reply:
yeah, I think we'll need either tlog or tsa here. Though, I suppose anything but CA could be optional.

Reply:
I think though that array type has a native nil, so I think the entries in it do not need to be pointers. Just need to add the // +optional above this.

Reply:
I am wondering now whether this conflicts with https://github.com/sigstore/policy-controller/pull/391/files#diff-83af2b17cb5361a5b3357d63c88e4965bfa07cb58917fbc0b88dd071cea39ccaR117 .

Reply:
So for consistency, I think it might be better to just specify the trusted certchain in one place (trustroot) rather than bake it into each CIP, just like for Fulcio/Rekor we don't bake the certs/keys in.

Reply:
re having the underscore in the name, that's not right :) For the JSON fields, the convention is camelcase.

Reply:
I think in the fullness of time the URLs from the keyless/ctlog might just go away and you just refer to the trust root, since the entries in the TrustRoot have the URLs necessary to construct the clients as well as all the other necessary bits.

Reply:
> So for consistency, I think it might be better to just specify the trusted certchain in one place (trustroot) rather than bake it into each CIP, just like for Fulcio/Rekor we don't bake the certs/keys in.

But we bake in the keys, and ctlog references 🤔 .

Reply:
Yeah, that's why I say in fullness of time :) I think CTLog / Fulcio URL will go away and they will just refer to the TrustRoot.